src/lints/function_export_name_changed.ron

```diff
@@ -0,0 +1,55 @@
+SemverQuery(
+    id: "function_export_name_changed",
+    human_readable_name: "function's export name has changed",
+    description: "A function's ABI name with #[no_mangle] or #[export_name = \"name\"] has changed",
+    required_update: Major,  // Not specified in rust semver doc? most similar is for #[repr] changes: https://doc.rust-lang.org/cargo/reference/semver.html#type-layout
+    reference_link: Some("https://doc.rust-lang.org/reference/abi.html#the-no_mangle-attribute"),  // TODO
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        export_name @output @tag
+                    }
+                }
+            }
+            
+            current {
+                item {
+                    ... on Function {
+                        name @output
+
+                        export_name @filter(op: "!=", value: ["%export_name"]) @filter(op: "!=", value: ["$null"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+                
+                item @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                    ... on Function {
+                        export_name @filter(op: "=", value: ["%export_name"])
+                    }
+                }
```

Comment:
It might be a good idea to write the lint for "export name no longer exists" first, just to get us a bit more test coverage and help us decide how to split the responsibilities between this lint and that one.

Here's what I'm worried about.

If both this lint and that other one enforce that the export name doesn't exist anymore, then it sounds like this lint is just a stricter version of that other one -- so whenever this one triggers, the other one will too. We want to avoid that.

So maybe we reshape this lint a bit to say "a function in the public API has changed its export name in the ABI" instead. So the invariant that is broken (and that we're reporting) is "call `func()` via the API, and `export_func()` via the ABI, those do the same thing because they are the same function." If the ABI export name changes to something else, then even if the ABI name `export_func()` continues to exist on some other function, this is likely worth flagging because it's a possible bug.

If we do end up taking that route, then I think:
- This block I've highlighted becomes unnecessary.
- The other lint doesn't need to look at `importable_path` at all, and can just focus on `export_name` alone.

What do you think?

Reply:
Yeah, I was thinking about that when I was writing it - it was pretty much the same query except checking if the current `export_name` *is* null. If we don't need granular feedback between missing vs changed, they can be easily combined. Do you think it's necessary?

Reply:
I'm not at my computer right now but I can push the other lint to a branch when I get a chance